File: backend-inmobiliaria/src/main/java/backend_inmobiliaria/service/ContactService.java
package backend_inmobiliaria.service;

import backend_inmobiliaria.dto.ContactDto;
import backend_inmobiliaria.dto.ContactResponseDto;
import backend_inmobiliaria.entity.ContactMessage;

import java.util.List;

public interface ContactService {

    void sendMessage(ContactDto dto);
    List<ContactResponseDto> getAllMessages();
    ContactResponseDto getMessageById(Long id);
    void markAsResponded(Long id);
    void deleteMessage(Long id);


}

File: frontend/src/services/ContactService.js
import api from "../api/api";


export const sendMessage = async (contactDto) => {
  const { data } = await api.post("/api/contacts", contactDto);
  return data;
};

// Admin
export const getAllMessages = async () => {
  const { data } = await api.get("/api/contacts");
  return data;
};

export const getMessageById = async (id) => {
  const { data } = await api.get(`/api/contacts/${id}`);
  return data;
};

export const markAsResponded = async (id) => {
  const { data } = await api.put(`/api/contacts/${id}/responded`);
  return data;
};

export const deleteMessage = async (id) => {
  const { data } = await api.delete(`/api/contacts/${id}`);
  return data;
};